### ISSUE TITLE
readNetFromTensorflow don't work

### ISSUE THREAD

**Issue by thallard** [opencv/opencv#21769]
##### System information (version)

- OpenCV => 4.5.5.64
- Operating System / Platform => Windows 11 64 Bit
- Compiler => Python 3.10.2

##### Detailed description

When I want to load my ``saved_model.pb`` which is on the same directory with ``cv.dnn.readNetFromTensorflow('saved_model.pb')``, I got this error :
``cv2.error: OpenCV(4.5.5) D:\a\opencv-python\opencv-python\opencv\modules\dnn\src\tensorflow\tf_io.cpp:42: error: (-2:Unspecified error) FAILED: ReadProtoFromBinaryFile(param_file, param). Failed to parse GraphDef file: saved_model.pb in function 'cv::dnn::ReadTFNetParamsFromBinaryFileOrDie'``

I saw everyone use already trained model and nobody train its own model for use it in OpenCV. I don't see anything related with my error on internet or to load a custom model in OpenCV. :/

##### Steps to reproduce

```model = tf.keras.models.Sequential([
  tf.keras.layers.Rescaling(1./255),
  tf.keras.layers.Conv2D(32, 3, padding='same', activation='relu'),
  tf.keras.layers.MaxPooling2D(),
  tf.keras.layers.Conv2D(64, 3, padding='same', activation='relu'),
  tf.keras.layers.MaxPooling2D(),
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dense(2)
])

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

model.fit(
  train_ds,
  validation_data=val_ds,
  epochs=10
)

model.save('./')
model = cv.dnn.readNetFromTensorflow('saved_model.pb')
``` 

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
   
 - [x] I updated to the latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by alalek** [comment]
> Failed to parse GraphDef file

This usually means that passed file doesn't exist. Or have malformed content.
Please double check file path and its content (e.g. through https://netron.app/ )

**Comment by ilknurakcay** [comment]
Did you find any solution?

**Comment by thallard** [comment]
Yes, my `saved_model.pb` was wrong, I used a wrong algorithm, I just changed to an other and it worked!

**Comment by avinashsharma080** [comment]
> > 
> 
> How did you fix it? i'm having same issue
cv2.dnn.readNetFromTensorflow(PATH_TO_CKPT, PATH_TO_PBTXT) 

you also need to attach the labels file alongwith the graph file
